pkg/operator/controller/ingress/status.go

```diff
@@ -641,43 +692,42 @@ func formatConditions(conditions []*operatorv1.OperatorCondition) string {
 	return formatted
 }
 
-// computeIngressProgressingCondition computes the ingresscontroller's
-// "Progressing" status condition, which indicates if the ingresscontroller's
-// current state matches its desired state.
-func computeIngressProgressingCondition(conditions []operatorv1.OperatorCondition, ic *operatorv1.IngressController, service *corev1.Service, platform *configv1.PlatformStatus) operatorv1.OperatorCondition {
-	condition := operatorv1.OperatorCondition{
-		Type:   operatorv1.OperatorStatusTypeProgressing,
-		Status: operatorv1.ConditionFalse,
+// computeIngressProgressingCondition computes the IngressController's current Progressing status state
+// by inspecting the following:
+// 1) the Deployment Replicas Updated condition of the IngressController
+// 2) the LoadBalancer Progressing condition of the IngressController
+// The IngressController is judged NOT Progressing only if all 2 conditions are true; otherwise
+// it is considered to be Progressing.
+func computeIngressProgressingCondition(conditions []operatorv1.OperatorCondition) operatorv1.OperatorCondition {
+	expected := []expectedCondition{
+		{
+			condition: IngressControllerLoadBalancerProgressingConditionType,
+			status:    operatorv1.ConditionFalse,
+		},
```

Comment:
You might need to add the `ifConditionsTrue` clause when defining `expectedCondition` for `IngressControllerLoadBalancerProgressingConditionType`, as we needed to for `LoadBalancerReadyIngressConditionType` in:
https://github.com/openshift/cluster-ingress-operator/blob/master/pkg/operator/controller/ingress/status.go#L257